Backend/TimeTracker.Api/Controllers/AuthController.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using TimeTracker.Api.Database;
+using TimeTracker.Api.Database.Models;
+using TimeTracker.Api.DTOs;
+using TimeTracker.Api.Helpers;
+
+namespace TimeTracker.Api.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class AuthController : ControllerBase
+    {
+        MainDb database;
+        IConfiguration configuration;
+        AuthHelper authHelper;
+
+        public AuthController(MainDb db, IConfiguration config, AuthHelper helper)
+        {
+            database = db;
+            configuration = config;
+            authHelper = helper;
+        }
+
+        /// <summary>
+        /// Endpoint to login the user
+        /// </summary>
+        /// <param name="loginData">This holds the username and password of the attempted login</param>
+        /// <returns>A response with an access key and a refresh key</returns>
+        [Route("/Login")]
+        [HttpPost]
+        public async Task<GenericResponseDTO<AccessKeysDTO>> Login(UserDTO loginData)
+        {
+            try
+            {
+                // Get user with a matching username and password hash
+                var hashedPassword = authHelper.GetPasswordHash(loginData.Password, configuration);
+                var curUser = await database.Users.FirstOrDefaultAsync(u => 
+                    u.Email == loginData.Email && u.Password.SequenceEqual(hashedPassword)
+                );
+
+                // If there was not a matching user then return an error
+                if(curUser == null)
+                {
+                    return new GenericResponseDTO<AccessKeysDTO>()
+                    {
+                        Success = false,
+                        Message = "Invalid username or password"
+                    };
+                }
+
+                // Generate the JWT and Refresh tokens and save the refresh token to the db
+                var accessToken = authHelper.GenerateJSONWebToken(curUser, configuration);
+                var refreshToken = authHelper.GenerateRefreshToken();
+
+                var userRefreshToken = new RefreshToken()
+                {
+                    Token = refreshToken,
+                    User = curUser
+                };
+
+                await database.AddAsync(userRefreshToken);
+                await database.SaveChangesAsync();
+
+                return new GenericResponseDTO<AccessKeysDTO>() 
+                {
+                    Success = true,
+                    Data = new AccessKeysDTO()
+                    {
+                         AccessToken = accessToken,
+                         RefreshToken = refreshToken
+                    }
+                };
+            }
+            catch
+            {
+                return new GenericResponseDTO<AccessKeysDTO>() 
+                {
+                    Success = false,
+                    Message = "An unknown error has occured"
```

Comment:
Do we want to pass back the exception to give a more in-depth error message?

Reply:
I wouldn't want to pass back the exception because in some cases exception messages can contain sensitive information. Any other thoughts?

Reply:
The main issue where this would happen is if there was a connection issue with the database.

Reply:
What are some possible errors that might be caught in this statement?

Reply:
Mainly database errors, but its really just a catch-all to make sure we return an OK response back to the user